bgcval2/functions/standard_functions.py

```diff
@@ -0,0 +1,199 @@
+#
+# Copyright 2014, Plymouth Marine Laboratory
+#
+# This file is part of the bgc-val library.
+#
+# bgc-val is free software: you can redistribute it and/or modify it
+# under the terms of the Revised Berkeley Software Distribution (BSD) 3-clause license. 
+
+# bgc-val is distributed in the hope that it will be useful, but
+# without any warranty; without even the implied warranty of merchantability
+# or fitness for a particular purpose. See the revised BSD license for more details.
+# You should have received a copy of the revised BSD license along with bgc-val.
+# If not, see <http://opensource.org/licenses/BSD-3-Clause>.
+#
+# Address:
+# Plymouth Marine Laboratory
+# Prospect Place, The Hoe
+# Plymouth, PL1 3DH, UK
+#
+# Email:
+# ledm@pml.ac.uk
+#
+
+"""
+.. module:: standard_functions
+   :platform: Unix
+   :synopsis: This module is a dictionairy of functions that can be applied to the data.
+       The expectation is that users will either add to this list.
+.. moduleauthor:: Lee de Mora <ledm@pml.ac.uk>
+
+"""
+import numpy as np
+from ..bgcvaltools.dataset import dataset
+
+
+#####
+# This is the algorithm that applied the evaluation functions below.
+def extractData(nc, details,key = ['',],debug=False):
+    """     
+    This extracts the data the the netcdf based on the instructions from details dictionairy.
+    If you want to do something funky to the data before plotting it, just create a new convert function in this directory.
+    Exampple details dict: 
+        {'name': 'Chlorophylla', 'vars':['Chlorophylla',], 'convert': std_functions['div1000'],'units':'ug/L'}
+    """
+    if isinstance(details,dict): 
+          if debug: print("extractData: details is a dict", list(details.keys()))
+    else:
+         raise AssertionError("std_functions:\textractData:\t Details is not a dictionairy!")        
+     
+    #####
+    # Finding key word arguments (kwargs):
+    # We define these as extra items in the details dict, and they can literally be anything
+    # that the convert function needs.
+    kwargs={}
+    for k in list(details.keys()):
+        if k in ['name','vars','convert', 'units']: continue
+        kwargs[k] = details[k]
+         
+    #####
+    # Apply pre-specified convert function.
+    xd = details['convert'](nc,details['vars'], **kwargs)
+    
+    #####
+    # Return masked array of the loaded data.
+    return np.ma.array(xd)
+
+
+####
+# Some functions for maniulating data:
+def NoChange(nc,keys):    
+    """ 
+    Loads keys[0] from the netcdf, but applies no change.
+    """
+    return nc.variables[keys[0]][:]
+
+
+def N2Biomass(nc,keys):    
+    """ 
+    Loads keys[0] from the netcdf, but multiplies by 79.572 (to convert Nitrogen into biomass).
+    """
+    return nc.variables[keys[0]][:]* 79.573
+
+
+def KtoC(nc,keys):        
+    """ 
+    Loads keys[0] from the netcdf, and converts from Kelvin to Celcius.
+    """
+    return nc.variables[keys[0]][:] - 273.15
```

Comment:
why do you keep expanding the data array with [:] - am I missing something here? It's all the same with or without [:]

Reply:
I'm not sure. nc is a bgcvaltools.dataset, not a netCDF4 dataset, so there may be some old reason for it, though I can't see it know.

Reply:
OK, no sweat